### DISCUSSION TITLE
Node resource scheduling.

### DISCUSSION THREAD

**Discussion by become-nice** [milvus-io/milvus#25371] [Q&A and General discussion]
I have a milvus cluster. When I use vectorDBBench to test, I find that there is a big difference in the execution time of two index_nodes to build the index. The following is an example. One node takes 1 hour, and one node only takes 30 minutes. Whether milvus can circumvent this situation.
![WechatIMG1576](https://github.com/milvus-io/milvus/assets/56624819/4de7b051-8bac-4c93-b583-b7a23bf9b086)
![WechatIMG1577](https://github.com/milvus-io/milvus/assets/56624819/f9e938ea-7ddf-4def-b557-425896d996fa)

**Comment by czs007** [comment]
@xiaocai2333

**Comment by xiaocai2333** [comment]
how much data of your test? And what's the index type?

**Reply by become-nice** [reply]
![111](https://github.com/milvus-io/milvus/assets/56624819/05e2c832-3deb-4163-be52-edb52d8e971e)

**Reply by xiaocai2333** [reply]
how many the segments of your cluster? you can get it of your cluster monitoring. I speculate that there might be exactly 3 segments, and it takes 30 minutes to build an index for one segment. Indexnode1 has built indexes for two segments.

**Reply by become-nice** [reply]
How can I get the number of segment?

**Reply by xiaocai2333** [reply]
can you  upload the indexnode logs here?

**Comment by xiaofan-luan** [comment]
> I have a milvus cluster. When I use vectorDBBench to test, I find that there is a big difference in the execution time of two index_nodes to build the index. The following is an example. One node takes 1 hour, and one node only takes 30 minutes. Whether milvus can circumvent this situation. ![WechatIMG1576](https://user-images.githubusercontent.com/56624819/251374127-4de7b051-8bac-4c93-b583-b7a23bf9b086.jpeg) ![WechatIMG1577](https://user-images.githubusercontent.com/56624819/251374176-f9e938ea-7ddf-4def-b557-425896d996fa.jpeg)

this is kind of based on the segment size.  smaller segment index build speed is faster and that is as expected.
to figure out the issue, you need to check out the log about which segment is building at that period and what is its size.
